src/Mail.php

```diff
@@ -19,9 +19,7 @@ public static function wp_mail(array $message) {
     $from_name = get_bloginfo('name');
     $from_email = get_option('admin_email');
     // Ensure that emails from staging site instances can still be identified.
-    if (isset($_SERVER['SERVER_NAME'])) {
-      $from_email = strtok($from_email, '@') . '@' . str_replace('www.', '', $_SERVER['SERVER_NAME']);
-    }
+    $from_email = strtok($from_email, '@') . '@' . str_replace('www.', '', parse_url(network_home_url(), PHP_URL_HOST));
```

Comment:
We should use the regular `site_url()` here, as `network_home_url()` returns the domain/URL of the operating network and not the current site itself.